### DISCUSSION TITLE
Can we not now use Variadic types for Node.js util.promisify.? "[@types/node]"

### DISCUSSION THREAD

**Discussion by KpjComp** [DefinitelyTyped/DefinitelyTyped#56604] [Issues with a @types package]
[@types/node]

Since version 4.0 of Typescript there has been support for variadic types, and currently `util.promisify` is using the multiple overloads  T1, T2, T3....... TResult  hack due to spread syntax and last param problem.

But now that we have variadic type in TS, is it worth changing this to be less hacky?.

Technically the current implementation is not technically complete, as it's limited to 5 parameters, but in theory there is nothing stopping anybody promisifying 6,7,+ params, so it would help with this too.

So for example using variadic type you could simplify and remove all the overloads with this ->.
```typescript
function promisify<T extends unknown[], TResult>
   (fn: (...args:[...T, (err:any, r?:TResult) => void]) => void):(...args:T) 
   => Promise<TResult> 
```

And a working Typescript Playground showing the types working etc.

[TS Playground](https://www.typescriptlang.org/play?strict=false&noImplicitAny=false&noStrictGenericChecks=true&module=1&noImplicitUseStrict=false&suppressExcessPropertyErrors=false&suppressImplicitAnyIndexErrors=false&keyofStringsOnly=false&ts=4.0.5&ssl=1&ssc=1&pln=1&pc=145#code/GYVwdgxgLglg9mABABwE5wLYwM42ATwB4AVRAUwA8oywATbRcAazDgHcwBtAXQBpFiAJTLYQAGygA+ABTAwALkTSAdKoCGqAObZ5nVcuL8lZVKnlqw+fqgD88oSPFQAlIgC8kxADc4MWt1cPb19aZ3kVdS0dYkDPAAV0LGwyEmFRCU8AbwAoRERUMigQVCQI5Q1tWMQcvLyCopLEMDI2RATMHDJpaQLsODEvMmsyACsyaCqa2ry5MorsfmkTVGH0l3cs3OntvGNTVwKx6CX9re3asjFk-JF+wZ7HCWcAbjPagF8Xt8+t9+y-7IAekB1GwUGwqmyoEgsAQiCgAEZpGp5GDUDAwJp+AAjVFQdGY-gQXF7MwAUVMcFQiAAPk1xGJVk48QTNFUfH5XJlEMTpGAGfwAAYAEkyaneiAA1IhRdj3oLnH8IAgwShEe4UIkcHh8NJEV8oeBoPAkFAAEzI+T8jDYkw4lkYrE8kkncmU6l0-liRk3NYOzHskJc518gWIbHKArIMhqKDI5yK7LKsCq5DmjVoDq4Ah6s0G6HGuFQADMluttpW4atIBtdudild8gp6A99O9TIk1drqEDnOqIa9PrUUvDieTqZLGa12d1JfzRthpoALJbsXB+jGwESXcsm+7aW2xL3Qv3dvGBy1EM2qdIAOTX1C3hPPchXMgDhljlVQNVLqdZnU9SXA1DRhE14REONMwweR2iSFILHwSQuS2aC3mUKAAAsaB6DYeRVDdlDEOBNDvXonFvawE3QiBYwgTCljw8dCOWG9b1Yx9+DIZ9-mybJQSgxE7zUSjEFvbEny+ATpDTC0l34W9sO9OBJNeaS01LBEEX4BE8ykyCZKgFd8RAbj9LBQyV2ANQ32fIA)

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
I'm afraid not soon, even if there are a lot of people that would love to rewrite stuff here and there. 3.7 is the minimum version against which DT tests are run. Node types are too large and complex to maintain types versioning on the other hand.
